lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -648,18 +648,39 @@ HdDirtyBits HdVP2Mesh::_PropagateDirtyBits(HdDirtyBits bits) const
         bits |= DirtySelectionHighlight;
     }
 
-    // Propagate dirty bits to all draw items.
-    for (const std::pair<TfToken, HdReprSharedPtr>& pair : _reprs) {
-        const HdReprSharedPtr& repr = pair.second;
-        const auto&            items = repr->GetDrawItems();
+    if (bits & HdChangeTracker::AllDirty) {
+        // RPrim is dirty, propagate dirty bits to all draw items.
+        for (const std::pair<TfToken, HdReprSharedPtr>& pair : _reprs) {
+            const HdReprSharedPtr& repr = pair.second;
+            const auto&            items = repr->GetDrawItems();
 #if HD_API_VERSION < 35
-        for (HdDrawItem* item : items) {
-            if (HdVP2DrawItem* drawItem = static_cast<HdVP2DrawItem*>(item)) {
+            for (HdDrawItem* item : items) {
+                if (HdVP2DrawItem* drawItem = static_cast<HdVP2DrawItem*>(item)) {
 #else
-        for (const HdRepr::DrawItemUniquePtr& item : items) {
-            if (HdVP2DrawItem* const drawItem = static_cast<HdVP2DrawItem*>(item.get())) {
+            for (const HdRepr::DrawItemUniquePtr& item : items) {
+                if (HdVP2DrawItem* const drawItem = static_cast<HdVP2DrawItem*>(item.get())) {
 #endif
-                drawItem->SetDirtyBits(bits);
+                    drawItem->SetDirtyBits(bits);
+                }
+            }
+        }
+    } else {
+        // RPrim is clean, find out if any drawItem about to be shown is dirty:
```

Comment:
What's the performance impact of this change?

Reply:
No performance impact. I checked and vertex buffers are correctly shared and are updated only once when changing a sphere radius. The geometries get updated for each displayed render item, but only as necessary to clean the dirty bits found in the reprs.